docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,451 @@ title = "Plugin migration guide"
+  d.logger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+// after
+package plugin
+
+import (
+  "context"
+
+  "github.com/grafana/grafana-plugin-sdk-go/backend"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+  "github.com/grafana/grafana-plugin-sdk-go/backend/log"
+  "github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+func NewSampleDatasource(_ backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	return &SampleDatasource{}, nil
+}
+
+type SampleDatasource struct{}
+
+
+func (d *SampleDatasource) Dispose() {
+	// Clean up datasource instance resources.
+}
+
+func (d *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+  log.DefaultLogger.Info("QueryData called", "request", req)
+  // logic for querying your datasource.
+}
+
+func (d *SampleDatasource) CheckHealth(_ context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+  log.DefaultLogger.Info("CheckHealth called", "request", req)
+  // The main use case for these health checks is the test button on the
+  // datasource configuration page which allows users to verify that
+  // a datasource is working as expected.
+}
+```
+
+### Unsigned backend plugins will not be loaded
+
+Before you can run a backend plugin on Grafana installations, you must sign the plugin. Grafana Labs strongly recommends you upload signed plugins since we cannot guarantee the integrity of unsigned plugins.
+
+For instructions on plugin signing, refer to the [Plugin signatures](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#sign-a-plugin) topic.
+
+You can still run and develop an unsigned plugin by running your Grafana instance in [development mode](https://grafana.com/docs/grafana/latest/administration/configuration/#app_mode).
+
+### Time series data can now be structured in a wide or many format.
+
+In Grafana 8.x, you can structure time series data in either `wide` or `many` formats. If your plugin supports time series data, we recommend that you support both formats.
+
+Grafana supported only the `many` format before. For example:
+
+| time                 | temperature |
+| -------------------- | ----------- |
+| 2021-05-08T07:31:45Z | 23          |
+| 2021-05-08T09:31:45Z | 25          |
+| 2021-05-08T11:31:45Z | 27          |
 
-It's written for:
+| time                 | humidity |
+| -------------------- | -------- |
+| 2021-05-08T07:31:45Z | 60       |
+| 2021-05-08T09:31:45Z | 55       |
+| 2021-05-08T11:31:45Z | 55       |
 
-- Plugin authors who want to migrate their plugins to Grafana 7.0+.
-- Plugin users who are using custom plugins and want to know whether they can upgrade to Grafana 7.0 without losing functionality.
+> Data is delivered as an array of data frames where the `time` field is repeated.
```

Comment:
I'm not sure if it's entirely clear here that these two tables represent multiple frames, with two fields each, in the same response (??). Though I'm not sure how else to structure it (i find data frames to be extremely difficult to visualise).

Perhaps instead it could be visualised as code? Dunno.

Reply:
Good point! Should I try to add it as `json` instead?

Reply:
Maybe as like Go or Javascript code that actually constructs the frames? I'm not sure if that's more obvious and easier to read? Just  throwing an idea out there.

Reply:
Resolving this one as this part got removed from the guide. ([More info and explanation](https://github.com/grafana/grafana/pull/38911#discussion_r708992021))